js/particle-modeler/index.js

```diff
@@ -43,17 +46,44 @@ export default class Interactive extends PureComponent {
     this.state = {
       interactive: models.interactive,
       model: model,
-      showNewAtom0: true,
-      showNewAtom1: true,
-      showNewAtom2: true,
+      showAtom0: true,
+      showAtom1: true,
+      showAtom2: true,
       deleteHover: false,
+      showRestart: false,
       ...authoredState
     };
 
     this.handleModelLoad = this.handleModelLoad.bind(this);
     this.addNewDraggableAtom = this.addNewDraggableAtom.bind(this);
     this.handleAuthoringPropChange = this.handleAuthoringPropChange.bind(this);
+    this.changeElementCount = this.changeElementCount.bind(this);
     this.freeze = this.freeze.bind(this);
+    this.restart = this.restart.bind(this);
+  }
+
+  componentWillMount() {
+    this.captureErrors();
+  }
+  captureErrors() {
+    window.onerror = (message, file, line, column, errorObject) => {
+      column = column || (window.event && window.event.errorCharacter);
+      var stack = errorObject ? errorObject.stack : null;
+
+      var data = {
+        message: message,
+        file: file,
+        line: line,
+        column: column,
+        errorStack: stack,
+      };
+      // If we want to log externally, use data, or grab on the console for more information
+      // console.log(data);
+      if (file.indexOf("lab.min.js") > -1) {
+        // likely indication of divergent lab model - offer a graceful restart button
```

Comment:
This would actually be really nice to add to the react-lab Lab component, so that you could just have an onError handler on the component. You may not want to have the added complication in this PR, but it would probably be widely useful, and clean up this code a bit.